util.go

```diff
@@ -1,11 +1,38 @@
 package tuna
 
 import (
+	"errors"
+	"io/ioutil"
+	"net/http"
 	"strings"
 
 	"github.com/nknorg/nkn/common"
 )
 
+const IP2CUrl = "https://ip2c.org/"
+
+type Location struct {
+	Code      string  `json:"Code"`
+	Country   string  `json:"Country"`
+	City      string  `json:"City"`
+	Latitude  float32 `json:"Latitude"`
+	Longitude float32 `json:"Longitude"`
+}
+
+func (l *Location) Empty() bool {
+	return *l == Location{}
+}
+
+type IPFilter struct {
+	Location  *Location
```

Comment:
We should support:
1. a list of country code
2. country blacklist

ipFilter: {
  allow: [
    {country: "US"},
    {country: "UK"},
    {ip: "1.2.3.4"}
  ],
  disallow: [
    {country: "XX"},
    {ip: "4.3.2.1"}
  ]
}